### PR TITLE
Add protocol for `IInfrastructureProvider`

### LINKED ISSUE #92: Add skeleton for `IInfrastructureProvider` interface as Protocol
## User stories
As a developer of the Interlocking component I want to know the skeleton of the interface IInfrastructureProvider in order to use the TraCIWrapper component.

## (Wiki) References
- https://github.com/BP2022-AP1/bp2022-ap1/wiki/Architecture#traciwrapper

## Acceptance Criteria
- [x] Add IInfrastructureProvider as protocol
- [x] Update wiki

### PR DESCRIPTION
**Pull request by SaturnHafen**
Fixes #92 

Description (what might a PO or reviewer want to know?)

## PR checklist

- [x] Acceptance criteria fulfilled
- [ ] ~~Additional features are tested~~
- [x] Docs (code, wiki & diagrams) updated
- [ ] ~~Breaking changes anounced~~
- [ ] Dev-branch has been merged into local branch to resolve conflicts
- [ ] Tests and linter have passed AFTER local merge
- [ ] Another dev reviewed and approved

### CODE REVIEW

File: src/interfaces.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+from collections.abc import Callable
+from typing import Protocol
+
+
+class IInfrastructureProvider(Protocol):
```
Comment (antonykamp) [inline src/interfaces.py:5]: Just a tiny question 😇 This is the same as https://github.com/simulate-digital-rail/interlocking/blob/yaramo-support/infrastructureprovider/infrastructureprovider.py, correct? A future class implements the `IInfrastructureProvider` Protocol and the `InfrastructureProvider` abstract class. Maybe we should make an exception here and only use the abstract one in the future.

Comment (SaturnHafen) [reply src/interfaces.py:5]: The protocol is implemented "automagically" when implementing/inheriting from the `InfrastructureProvider`. I would use this protocol for consistency with the rest of the project.

Comment (antonykamp) [reply src/interfaces.py:5]: I like the word _automagically_ 😂 
But you're right. As long as the abstract class doesn't change, we're good to go 🙌

**Review by antonykamp — APPROVED** [review]

**Comment by SaturnHafen** [comment]
The database test seems flaky :thinking: @antonykamp

**Pull request merged**